enumerables.rb

```diff
@@ -0,0 +1,154 @@
+    self
+  end
+
+  def my_select
+    if block_given?
+      arr = []
+      my_each { |element| arr.push(element) if yield(element) }
+      arr
+    else
+      to_enum(:my_select)
+    end
+  end
+
+  def my_all?(argm = nil)
+    if block_given?
+      my_each { |element| return false if yield(element) == false }
+      true
+    elsif argm.nil?
+      my_each { |n| return false if n.nil? || n == false }
+    elsif !argm.nil? && (argm.is_a? Class)
+      my_each { |n| return false if n.is_a?(argm) == false }
+    elsif !argm.nil? && argm.class == Regexp
+      my_each { |n| return false unless argm.match(n) }
+    else
+      my_each { |n| return false if n != argm }
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    if block_given?
+      my_each { |item| return true if yield(item) }
+      false
+    elsif arg.nil?
+      my_each { |n| return true if n }
+    elsif !arg.nil? && (arg.is_a? Class)
+      my_each { |n| return true if n.is_a?(arg) == true }
+    elsif !arg.nil? && arg.class == Regexp
+      my_each { |n| return true if arg.match(n) }
+    else
+      my_each { |n| return true if n == arg }
+    end
+    false
+  end
+
+  def my_none?(arg = nil)
+    if !block_given? && arg.nil?
+      my_each { |n| return false if n }
+      return true
+    end
+
+    if !block_given? && !arg.nil?
+
+      if arg.is_a?(Class)
+        my_each { |n| return false if n.class == arg }
+        return true
+      end
+
+      if arg.class == Regexp
+        my_each { |n| return false if arg.match(n) }
+        return true
+      end
+
+      my_each { |n| return false if n == arg }
+      return true
+    end
+
+    my_any? { |item| return false if yield(item) }
+    true
+  end
+
+  def my_count(arg = nil)
+    count = 0
+    if block_given?
+      to_a.my_each { |i| count += 1 if yield(i) }
+    elsif !arg.nil?
+      to_a.my_each { |i| count += 1 if i == arg }
+    else
+      to_a.my_each { |i| count += 1 if i }
+    end
+    count
+  end
+
+  def my_map(proc = nil)
+    return to_enum(:my_map) unless block_given?
+
+    result = []
+    my_each { |element| result << proc.call(element) } if block_given? && proc
+    my_each { |element| result << yield(element) } if proc.nil?
+    result
+  end
+
+  def my_inject(number = nil, symbol = nil)
+    raise LocalJumpError, 'No parameters', caller unless block_given?
+
+    if block_given?
+      accum = number
+      my_each do |item|
+        accum = accum.nil? ? item : yield(accum, item)
+      end
+      accum
+    elsif !number.nil? && (number.is_a?(Symbol) || number.is_a?(String))
+      accum = nil
+      my_each do |item|
+        accum = accum.nil? ? item : accum.send(number, item)
+      end
+      accum
+    elsif !symbol.nil? && (symbol.is_a?(Symbol) || symbol.is_a?(String))
+      accum = number
+      my_each do |item|
+        accum = accum.nil? ? item : accum.send(symbol, item)
+      end
+      accum
+    end
+  end
```

Comment:
## #my_inject should have the following behaviors:
- [ ]  when a symbol is specified without an initial value combines each element of the collection by applying the symbol as a named method
- [ ] when a symbol is specified without an initial value combines each element of the collection by applying the symbol as a named method:range
- [ ] when a symbol is specified with an initial value combines each element of the collection by applying the symbol as a named method
- [ ] when a symbol is specified with an initial value combines each element of the collection by applying the symbol as a named method::range